[Geom] TGeoVGConverter: Introduce shape-type filtering for conversion process

ihrivnac opened the pull request:
- Added functions `SelectShapeType()` and `ExcludeShapeType()`. Conversion is performed only on selected shapes (if set), and excluded shapes are always skipped.

- TGeoVGShape: Disabled conversion for problematic edge cases not covered by exclusion: 'TGeoBBox' with shifted origin and 'TGeoScaledShape'.

# This Pull request:

## Changes or fixes:
- Added `SelectShapeType()` and `ExcludeShapeType()` functions to `TVirtualGeoConverter` and  `TGeoVGConverter`
- In `TGeoVGShape::Convert()`: Disabled conversion for 'TGeoBBox' with shifted origin and 'TGeoScaledShape'

## Checklist:

- [ x] tested changes locally
- [ x] updated the docs (if necessary)

This PR fixes #

agheata commented on geom/vecgeom/src/TGeoVGConverter.cxx:
```diff
@@ -65,3 +115,20 @@ void TGeoVGConverter::ConvertGeometry()
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Exclude shape(s) from conversion.
```
Maybe add in the comment: "Excluded types have precedence in case a type is also selected."

ihrivnac commented:
Hello @agheata ,

I have addressed your comments; please, check, and resolve threads which you find ok.
There is also updated `TGeoCompositeShape` where setting the shape type was missing in one ctor.
